File: Resources/Code/Data-Structure/Search/Binary-Search/python/BinarySearch.py
class Search:
  def __init__(self, array_of_numbers):
    self.array_of_numbers = array_of_numbers

  def binary(self, number_to_find):
    last_pos=len(self.array_of_numbers)-1
    first_pos=0 
    mid=0

    while (first_pos<=last_pos):
        mid = (first_pos + last_pos)//2

        if self.array_of_numbers[mid] == number_to_find:
            return True
        elif self.array_of_numbers[mid] > number_to_find:
            last_pos = mid - 1
        else:
            first_pos = mid + 1
    return False

array_of_numbers = [1, 3, 5, 9, 13]
search = Search(array_of_numbers)
number_to_find = int(input('What number do you want find?: '))
print(search.binary(number_to_find))

File: Software/Data-Structure/Search/Binary-Search/cpp/BinarySearch.cpp
#include <iostream>

using namespace std;

int main()
{
    int array_of_numbers[5] = { 1, 3, 5, 9, 13 };  
    int number_to_find = 0;
    
    cout<<"What number do you want find?: ";
    cin>>number_to_find;

    int last_pos = sizeof(array_of_numbers)/sizeof(int) - 1;
    int mid = 0, first_pos = 0;

    while (first_pos<=last_pos) {
        mid = (last_pos + first_pos)/2;
        if (array_of_numbers[mid] == number_to_find) {
            cout<<"True";
            return 0;
        }
        else if (array_of_numbers[mid] > number_to_find) {
            last_pos = mid-1;
        }
        else {
            first_pos = mid+1;
        }
    }
        
    cout<<"False";
    return 0;
}
